libs/libc/stream/lib_mtdoutstream.c

```diff
@@ -342,18 +330,22 @@ int lib_mtdoutstream_open(FAR struct lib_mtdoutstream_s *stream,
       return -EINVAL;
     }
 
-  stream->cache = lib_malloc(stream->geo.erasesize);
-  if (stream->cache == NULL)
+#ifdef CONFIG_MTD_BYTE_WRITE
```

Comment:
Please include a comment here, to explain the case of MTD_BYTE_WRITE enable and the case it is not enabled. It will help people to understand the goal of this part of the code